semantic/package.go

```diff
@@ -0,0 +1,74 @@
+package semantic
+
+import (
+	"github.com/pkg/errors"
+)
+
+// Package represents the type and name of a package.
+type Package struct {
+	Name string
+	Type PolyType
+}
+
+// CreatePackage constructs a Package from the node. The node must contain a Program node
+// with a valid PackageClause
+func CreatePackage(n Node, importer Importer) (Package, error) {
```

Comment:
Will packages and imports be replacing externs completely? If so, should this function specifically take a `Program` type instead of `Node`?

Reply:
They will not, atleast I do not expect them to. There are some builtin values in the top level namespace that are not part of any package. Maybe we can model those as a package in the future but not yet.